internal/manifests/collector/podmonitor.go

```diff
@@ -63,6 +63,7 @@ func PodMonitor(params manifests.Params) (*monitoringv1.PodMonitor, error) {
 				MatchLabels: map[string]string{
 					"app.kubernetes.io/managed-by": "opentelemetry-operator",
 					"app.kubernetes.io/instance":   fmt.Sprintf("%s.%s", params.OtelCol.Namespace, params.OtelCol.Name),
+					"app.kubernetes.io/name":       "otel-collector-monitoring",
```

Comment:
I don't think this name is accurate, as mikolaj said we need to use manifestutils here to select the right instance. ([ex](https://github.com/open-telemetry/opentelemetry-operator/blob/ca0b9498f48fdf6bba3687d3a193bed685eba2e9/internal/manifests/manifestutils/labels.go#L67-L69)) Further, I don't believe the name for the podmonitor will contain the -monitoring suffix as we only add that for the monitoring service.